plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,577 @@
+
+SUCCESS_EXPORT_MSG = "Successfully exported the license."
+SUCCESS_DELETE_MSG = "Successfully deleted the license."
+SUCCESS_IMPORT_MSG = "Successfully imported the license."
+FAILURE_MSG = "Unable to {operation} the licens with id {license_id}."
+
+IGNORE_CERTIFICATE_WARNING = {"off": "Off", "on": "On"}
+PROXY_SUPPORT = {"off": "Off", "default_proxy": "DefaultProxy", "parameters_proxy": "ParametersProxy"}
+PROXY_TYPE = {"http": "HTTP", "socks": "SOCKS"}
+
+
+class DeleteLicense():
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        license_id = module.params.get('license_id')
+        delete_license_url = f"/redfish/v1/LicenseService/Licenses/{license_id}"
+        delete_license_status = self.idrac.invoke_request(delete_license_url, 'DELETE')
+        status = delete_license_status.status_code
+        share_type = module.params.get('share_parameters').get('share_type')
+        if status == 204:
+            module.exit_json(msg=SUCCESS_DELETE_MSG, changed=True)
+        else:
+            module.fail_json(FAILURE_MSG.format(operation=share_type, license_id=license_id), changed=False)
+        return delete_license_status
+
+
+class ExportLicense():
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        share_type = module.params.get('share_parameters').get('share_type')
+        license_id = module.params.get('license_id')
+        if share_type == "local":
+            export_license_status = export_license_local(self, module)
+        elif share_type in ["http", "https"]:
+            export_license_status = export_license_http(self, module)
+            job_status = get_job_status(self, export_license_status)
+        elif share_type == "cifs":
+            export_license_status = export_license_cifs(self, module)
+            job_status = get_job_status(self, export_license_status)
+        elif share_type == "nfs":
+            export_license_status = export_license_nfs(self, module)
+            job_status = get_job_status(self, export_license_status)
+        status = export_license_status.status_code
+        if share_type in ["http", "https", "cifs", "nfs"]:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_EXPORT_MSG, changed=True, job_details=job_status)
+            else:
+                module.fail_json(msg=FAILURE_MSG.format(operation=share_type, license_id=license_id), changed=False, job_details=job_status)
+        else:
+            if status in [200, 202]:
+                module.exit_json(msg=SUCCESS_EXPORT_MSG, changed=True)
+            else:
+                module.fail_json(msg=FAILURE_MSG.format(operation=share_type, license_id=license_id), changed=False)
+        return export_license_status
+
+
+def get_job_status(self, export_license_status):
```

Comment:
check utils.py if we have any common function for getting job_status, if not better to add this function there, as it can be used in future module, make sure to update export_licence_status to generic variable name for idrac response.

Reply:
This function uses 'idrac_redfish_job_tracking' for tracking the job. It only handles the retrieval of the license job URI